ckanext/canada/assets/internal/canada_guess_mimetype.js

```diff
@@ -0,0 +1,140 @@
+window.addEventListener('load', function(){
+  $(document).ready(function() {
+
+    const lang = $('html').attr('lang').length > 0 ? $('html').attr('lang') : 'en';
+
+    // slightly different error messages for the front-end users.
+    // as the UX is different than the API users.
+    let failedGuessErrorMessage = 'Could not determine a resource format. Please supply a format below.';
+    let failedChoicesErrorMessage = 'Could not determine a valid resource format. Please supply a different format below.';
```

Comment:
I don't think we need error messages, we could leave the format drop-down blank and let the form validation tell them they need to choose something

Reply:
Yeah that makes sense. The logic method will just raise the error and then it will be added to the errors in the validator method.